bundle/config/validate/enum.go

```diff
@@ -0,0 +1,104 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/internal/validation/generated"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type enum struct{}
+
+func Enum() bundle.Mutator {
+	return &enum{}
+}
+
+func (f *enum) Name() string {
+	return "validate:enum"
+}
+
+func (f *enum) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	// Generate prefix tree for all enum fields.
+	trie := &dyn.TrieNode{}
+	for k := range generated.EnumFields {
+		pattern, err := dyn.NewPatternFromString(k)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("invalid pattern %q for enum field validation: %w", k, err))
+		}
+
+		err = trie.Insert(pattern)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to insert pattern %q into trie: %w", k, err))
+		}
+	}
+
+	err := dyn.WalkReadOnly(b.Config.Value(), func(p dyn.Path, v dyn.Value) error {
+		// If the path is not found in the prefix tree, we do not need to validate any enum
+		// fields in it.
+		pattern, ok := trie.SearchPath(p)
+		if !ok {
+			return nil
+		}
+
+		// Skip validation if the value is not set
+		if v.Kind() == dyn.KindInvalid || v.Kind() == dyn.KindNil {
```

Comment:
We already filter out everything but strings below, this check can be dropped, right?

Reply:
Good point, removed.